app/src/test/js/searchKeysCtrlTest.js

```diff
@@ -0,0 +1,48 @@
+describe('SearchKeysCtrl', function() {
+
+    beforeEach(module('esbMessageAdminApp', function($provide) {
+        $provide.value("EsbMessageService", mockService);
+    }));
+
+    var searchKeysCtrl, rootScope;
+
+    beforeEach(inject(function($rootScope, $controller, _$q_) {
+        rootScope = $rootScope;
+        searchKeysScope = $rootScope.$new();
+        $q = _$q_;
+        spyOn(mockService, "getSearchKeysTree").and.returnValue($q
+                .when(searchKeysSuccessResponse));
+        searchKeysCtrl = $controller('SearchKeysCtrl', {
+            '$scope' : searchKeysScope
+        });
+        rootScope.$apply();
+
+    }));
+
+    it('loads Tab', function() {
+        loadTest(searchKeysScope, searchKeysSuccessResponse);
+    });
+
+    it('crumbs', function() {
+        crumbTest(searchKeysScope, searchKeysSuccessResponse);
+    });
+
+    it('manages chidren', function() {
```

Comment:
`it("should populate crumbs with parent and child nodes when navigating to first child")`